newIDE/app/src/EventsSheet/EventsTree/index.js

```diff
@@ -239,35 +268,59 @@ type EventsTreeProps = {|
 
 // A node displayed by the SortableTree. Almost always represents an
 // event, except for the buttons at the bottom of the sheet.
-type SortableTreeNode = {
+export type SortableTreeNode = {
   eventsList: gdEventsList,
   event: ?gdBaseEvent,
   depth: number,
   disabled: boolean,
   indexInList: number,
+  nodePath: Array<number>,
+  children: Array<any>,
+  expanded: boolean,
+  key: number,
 
   // In case of nodes without event (buttons at the bottom of the sheet),
   // use a fixed height.
   fixedHeight?: ?number,
 };
 
+type State = {
+  treeData: Array<any>,
+  flatData: Array<gdBaseEvent>,
+  draggedNode: ?SortableTreeNode,
+  isScrolledTop: boolean,
+  isScrolledBottom: boolean,
+};
+
 /**
  * Display a tree of event. Builtin on react-sortable-tree so that event
  * can be drag'n'dropped and events rows are virtualized.
  */
-export default class ThemableEventsTree extends Component<EventsTreeProps, *> {
+export default class ThemableEventsTree extends Component<
+  EventsTreeProps,
+  State
+> {
   static defaultProps = {
     selection: getInitialSelection(),
   };
   _list: ?any;
   eventsHeightsCache: EventHeightsCache;
+  DragSourceAndDropTarget = makeDragSourceAndDropTarget<SortableTreeNode>(
+    eventsSheetEventsDnDType
+  );
+  DropTarget = makeDropTarget<SortableTreeNode>(eventsSheetEventsDnDType);
+  temporallyUnfoldedNodes: Array<SortableTreeNode>;
```

Comment:
Should it be `temporaryUnfoldedNodes`?

Reply:
I cannot believe Google translate failed me like this :(
<img width="672" alt="image" src="https://user-images.githubusercontent.com/81410437/170226261-8e44f2d2-515d-4f6b-a8f3-7b2e92967085.png">

Reply:
Ahah and I failed you with my suggestion!